wasm/jit/jit_amd64_test.go

```diff
@@ -1342,17 +1336,267 @@ func TestAmd64Builder_handleLe(t *testing.T) {
 				)
 				// Check the stack.
 				require.Equal(t, uint64(1), eng.stackPointer)
-				if tc.exp {
-					require.Equal(t, uint64(1), eng.stack[eng.stackPointer-1])
+				require.Equal(t, tc.x1 <= tc.x2, eng.stack[eng.stackPointer-1] == 1)
+			})
+		}
+	})
+}
+
+func TestAmd64Compiler_compileGe(t *testing.T) {
+	t.Run("int32", func(t *testing.T) {
+		for i, tc := range []struct {
+			x1, x2 int32
+			signed bool
+		}{
+			{x1: 100, x2: -1, signed: false},
+			{x1: -1, x2: -1, signed: false},
+			{x1: -1, x2: 100, signed: false},
+			{x1: 100, x2: 200, signed: true},
+			{x1: 100, x2: 100, signed: true},
+			{x1: 200, x2: 100, signed: true},
+		} {
+			t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
+				compiler := requireNewCompiler(t)
+				compiler.initializeReservedRegisters()
+				err := compiler.compileConstI32(&wazeroir.OperationConstI32{Value: uint32(tc.x1)})
+				require.NoError(t, err)
+				x1 := compiler.locationStack.peek()
+				err = compiler.compileConstI32(&wazeroir.OperationConstI32{Value: uint32(tc.x2)})
+				require.NoError(t, err)
+				x2 := compiler.locationStack.peek()
+				var o *wazeroir.OperationGe
+				if tc.signed {
+					o = &wazeroir.OperationGe{Type: wazeroir.SignedTypeInt64}
+				} else {
+					o = &wazeroir.OperationGe{Type: wazeroir.SignedTypeUint64}
+				}
+				err = compiler.compileGe(o)
+				require.NoError(t, err)
+
+				require.NotContains(t, compiler.locationStack.usedRegisters, x1.register)
+				require.NotContains(t, compiler.locationStack.usedRegisters, x2.register)
+				// To verify the behavior, we push the flag value
```

Comment:
we should have a TODO to make a utility that does this as it is everywhere. I think the rationale is we need to do this because we can't verify register state in a unit test, so we must move it to the stack. At least unless we have a virtual machine :D

Reply:
yep, I will raise an issue for tracking this!

Reply:
also added TODO comment: https://github.com/tetratelabs/wazero/pull/99/commits/cc5017a23aad8a6f072a2b8d018b1fb4cfb7e727